proposals/0235-add-result.md

```diff
@@ -54,11 +54,11 @@ URLSession.shared.dataTask(with: url) { (data, response, error) in
 While this code is only a few lines long, it exposes Swift's complete lack of automatic error handling for asynchronous APIs. Not only was the `error` forcibly unwrapped (or perhaps handled using a slightly less elegant `if` statement), but a possibly impossible scenario was created. What happens if `response` or `data` are `nil`? Is it even possible? It shouldn't be, but Swift currently lacks the ability to express this impossibility. Using `Result` for the same scenario allows for much more elegant code:
 
 ```swift
-URLSession.shared.dataTask(with: url) { (result: Result<(URLResponse, Data), Error>) in // Type added for illustration purposes.
+URLSession.shared.dataTask(with: url) { (result: Result<(response: URLResponse, data: Data?), Error>) in // Type added for illustration purposes.
```

Comment:
No, I've been informed that a non-optional `Data` is correct.  An `Error` is produced in case of a transport error and precludes the response and data.  If there's no transport failure, there will never be an `Error` and both the response and the data will be available, although of course the response may include a server status code reporting a higher-level error.

Reply:
Right, forgot about that. Fixed in 
https://github.com/apple/swift-evolution/pull/979/commits/ae16a16c414968939f9533800ff268fb611740c7.